### ISSUE TITLE
missing affine expression operations

### ISSUE THREAD

**Issue by jd-lara** [JuliaStochOpt/ParameterJuMP.jl#40]
There are a couple of operations missing

```Julia
model = ModelWithParams()
x = @variable(model)
y = @variable(model)
a = Parameter(model, 10)
b = Parameter(model, 11)

exp1 = x + y + a
exp2 = exp1 + b

MethodError: no method matching +(::Parameter, ::GenericAffExpr{Float64,Parameter})
Closest candidates are:
  +(::Any, ::Any, !Matched::Any, !Matched::Any...) at operators.jl:502
  +(::Parameter, !Matched::VariableRef) at /Users/jdlara/.julia/packages/ParameterJuMP/kVLVr/src/operators.jl:32
  +(::Parameter, !Matched::Parameter) at /Users/jdlara/.julia/packages/ParameterJuMP/kVLVr/src/operators.jl:36
  ...

Stacktrace:
 [1] +(::Parameter, ::ParameterJuMP.ParametrizedAffExpr{Float64}) at /Users/jdlara/.julia/packages/ParameterJuMP/kVLVr/src/operators.jl:48
 [2] +(::ParameterJuMP.ParametrizedAffExpr{Float64}, ::Parameter) at /Users/jdlara/.julia/packages/ParameterJuMP/kVLVr/src/operators.jl:115
 [3] top-level scope at In[30]:8
```

it will also useful to have a convert to promote and avoid this error 

```
MethodError: Cannot `convert` an object of type GenericAffExpr{Float64,VariableRef} to an object of type ParameterJuMP.ParametrizedAffExpr{Float64}
```

**Comment by jd-lara** [comment]
In https://github.com/JuliaStochOpt/ParameterJuMP.jl/blob/23b0250d8091dfd2d9a651e558399de6057f002d/src/operators.jl#L87

seems that the assumption is that JuMP takes care of that, but it doesn't

**Comment by jd-lara** [comment]
This is also missing 

```
MethodError: no method matching add_to_expression!(::ParameterJuMP.ParametrizedAffExpr{Float64}, ::VariableRef)
```

**Comment by jd-lara** [comment]
One more missing 

```
MethodError: no method matching add_to_expression!(::ParameterJuMP.ParametrizedAffExpr{Float64}, ::GenericAffExpr{Float64,VariableRef})
```

**Comment by joaquimg** [comment]
solved by #45

**Comment by jd-lara** [comment]
@joaquimg given some recent changes in the sums in JuMP when I do `Sum()` over a group of PGAE then I get his missing method error

```
add_to_expression!(::ParameterJuMP.DoubleGenericAffExpr{Float64,VariableRef,ParameterJuMP.Parameter}, ::ParameterJuMP.DoubleGenericAffExpr{Float64,VariableRef,ParameterJuMP.Parameter})
```
